File: cs50_submit/week2/readability/readability.c
#include <string.h>
#include <stdio.h>
#include <cs50.h>

float words (string input);
float sentences (string input);
void algorithim (float L, float S);
int main (void)
{
    float L = 0, S = 0;
    int gread = 0;
    string text = get_string("add your text : ");
    L = words(text);
    S = sentences(text);
    algorithim(L, S);
}

float words (string input)
{
    int letters = 0, word = 1, length = strlen(input);

    for (int i = 0; i < length; i++)
    {
        if (word >=100)
        {
            break;
        }
        else if (input[i] == ' ')
        {
            word++;
        }
        else
        {
            letters++;
        }
    }

    return ((float)letters / word) * 100;
}

float sentences (string input)
{
    int word = 0, sentence = 0, length = strlen(input);

    for (int i = 0; i < length; i++)
    {
        if (input[i] == ' ')
        {
            word++;
        }
        else if (input[i] == '.' || input[i] == '!' || input[i] == '?')
        {
            sentence++;
        }
    }

    return ((float)sentence / word) * 100;
}

void algorithim (float L, float S)
{
    int index = 0.0588 * L - 0.296 * S - 15.8;
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: 2025/repo/w6/sentimental-readability/readability.py
def main():
    # نأخذ النص من المستخدم
    text = input("Text: ")

    # نحسب عدد الحروف والكلمات والجمل
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # نحسب متوسط الحروف والجمل لكل 100 كلمة
    L = (letters / words) * 100
    S = (sentences / words) * 100

    # نحسب مؤشر Coleman-Liau
    index = 0.0588 * L - 0.296 * S - 15.8
    grade = round(index)  # نقرب الناتج لأقرب عدد صحيح

    # نطبع النتيجة حسب المطلوب
    if grade < 1:
        print("Before Grade 1")
    elif grade >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {grade}")


def count_letters(text):
    """تحسب عدد الحروف في النص"""
    count = 0
    for char in text:
        if char.isalpha():
            count += 1
    return count


def count_words(text):
    """تحسب عدد الكلمات في النص"""
    count = 1  # نبدأ بـ 1 لأن أول كلمة غالباً لا يسبقها مسافة
    for char in text:
        if char == ' ':
            count += 1
    return count


def count_sentences(text):
    """تحسب عدد الجمل في النص"""
    count = 0
    for char in text:
        if char in ['.', '!', '?']:
            count += 1
    return count


# استدعاء الدالة الرئيسية
main()
